tls/s2n_server_new_session_ticket.c

```diff
@@ -82,8 +82,10 @@ int s2n_server_nst_send(struct s2n_connection *conn)
     uint32_t lifetime_hint_in_secs =
             (conn->config->encrypt_decrypt_key_lifetime_in_nanos + conn->config->decrypt_key_lifetime_in_nanos) / ONE_SEC_IN_NANOS;
 
-    /* When server changes it's mind mid handshake send lifetime hint and session ticket length as zero */
-    if (!conn->config->use_tickets) {
+    /* When server changes it's mind mid handshake, or if session key used to encrypt session ticket
```

Comment:
This is probably a good time to introduce you to our compliance comments. We have a specific format we use to cite RFCs, and this is a good place for one since it's a strange case.
Here's an example of one; they get compiled into a report that lets us know how much of the RFC we're actually implementing: https://github.com/aws/s2n-tls/blob/main/tls/s2n_cipher_suites.c#L1100